examples/remote-sensing/helpers/processing_helpers.py

```diff
@@ -0,0 +1,203 @@
+        return self.fit(X).transform(X)
+
+
+class SlidingWindowVectorize(BaseEstimator, TransformerMixin):
+    """Sliding window for three-dimensional image.
+
+    Parameters
+    ----------
+    window_size : int
+        Size of the window.
+    overlap : int, default=0
+        Overlap between windows.
+    """
+
+    def __init__(self, window_size: int, overlap: int = 0):
+        assert window_size % 2 == 1, "Window size must be odd."
+        assert overlap >= 0, "Overlap must be positive."
+        assert overlap <= window_size//2, \
+            "Overlap must be smaller or equal than int(window_size/2)."
+        self.window_size = window_size
+        self.overlap = overlap
+
+    def fit(self, X: ArrayLike, y=None):
+        return self
+
+    def transform(self, X: ArrayLike):
+        """Transforms a multidimensional array (ndim=3) into a sliding window
+        view over the first two dimensions.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_rows, n_columns, n_features)
+            Input image
+
+        Returns
+        -------
+        Xnew : ndarray, shape (n_pixels, window_size**2, n_features)
+            Output array, with n_pixels =
+                (n_rows-window_size+1)(n_columns-window_size+1)//overlap^2
+        """
+        X = sliding_window_view(
+            X,
+            window_shape=(self.window_size, self.window_size),
+            axis=(0, 1),
+        )
+        if self.overlap is not None:
+            if self.overlap > 0:
+                X = X[::self.overlap, ::self.overlap]
+        else:
+            X = X[::self.window_size//2, ::self.window_size//2]
+            self.overlap = self.window_size//2
+
+        # Reshape to (n_pixels, n_samples, n_features) with
+        # n_pixels = axis0*axis1
+        # n_samples = axis3*axis_4
+        # n_features = axis2
+        X = X.reshape((-1, X.shape[2], X.shape[3]*X.shape[4]))
+        return X
+
+    def fit_transform(self, X: ArrayLike, y=None):
+        return self.fit(X).transform(X)
+
+
+class LabelsToImage(BaseEstimator, TransformerMixin):
+    """Predicted labels to image taking into account sliding windows.
+
+    Parameters
+    ----------
+    height : int
+        Height of the original image.
+    width : int
+        Width of the original image.
+    window_size : int
+        Size of the window.
+    """
+
+    def __init__(self, height: int, width: int,
+                 window_size: int, overlap: int = 0):
+        assert window_size % 2 == 1, "Window size must be odd."
+        assert overlap >= 0, "Overlap must be positive."
+        assert overlap <= window_size//2, \
+            "Overlap must be smaller or equal than int(window_size/2)."
+        self.height = height
+        self.width = width
+        self.overlap = overlap
+        self.window_size = window_size
+
+    def fit(self, X: ArrayLike, y=None):
+        return self
+
+    def transform(self, X: ArrayLike):
```

Comment:
Yes kinda, but it's very specific to one-dimensional output (1 label per pixel), whilst in my view we could use sliding windows to do any number of processing (for example just stopping at covariances: 1 covariance per window which is sometimes useful in PolSAR or InSAR applications, in which case recasting as one covariance per pixel would be a bit different).

I guess we could generalize the process to any number of dimensions (and we could apply that to sliding windows as well). But I didn't want to add more complexity to those examples since we only want some labels at the end. 

Also if I remember correctly, but might be wrong, I wanted to add it to the pipeline but the way Kmeans works doesn't allow to pass a `TransformerMixin` after to call `fit_transform` with this as the end of the pipeline.

Reply:
Ok. Let's leave generalization aside.

However, I suggest to move `LabelsToImage.transform()` into a new function `SlidingWindowVectorize.inverse_predict()`.

In this way, `LabelsToImage` can be deleted and we only use one class to process data.

Current code:
```
labels_pred = LabelsToImage(height, width, window_size).fit_transform(
    pipeline.named_steps["kmeans"].labels_
)
results[pipeline_name] = labels_pred
```
can be replaced by
```
preds = pipeline.named_steps["kmeans"].labels_
results[pipeline_name] = pipeline.named_steps["sliding_window"].inverse_predict(preds)
```

For me, we can merge the branch now, but I think it's worth finalizing the image processing classes (code and documentation), in order to quickly be able to add them to the pyRiemann API.

Reply:
Ok I change that in the way you propose. It is indeed simpler!